contracts/Core/BlockManager.sol

```diff
@@ -38,21 +65,26 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
         randomNoProvider = IRandomNoProvider(randomNoManagerAddress);
     }
 
-    // elected proposer proposes block.
-    //we use a probabilistic method to elect stakers weighted by stake
-    // protocol works like this.
-    //select a staker pseudorandomly (not weighted by anything)
-    // that staker then tosses a biased coin.
-    //bias = hisStake/biggestStake. if its heads, he can propose block
-    // end of iteration. try next iteration
-    // note that only one staker or no stakers selected in each iteration.
-    // stakers elected in higher iterations can also propose hoping that
-    // stakers with lower iteration do not propose for some reason
-
-    /// @dev The IDs being passed here, are only used for disputeForNonAssignedCollection
-    /// for delegator, we have seprate registry
-    /// If user passes invalid ids, disputeForProposedCollectionIds can happen
-
+    /**
+     * @notice elected proposer proposes block.
+     * we use a probabilistic method to elect stakers weighted by stake
+     * protocol works like this.
+     * select a staker pseudorandomly (not weighted by anything)
```

Comment:
```
 select a staker pseudorandomly (not weighted by anything)
 that staker then tosses a biased coin
 ```
hmm is this true anymore ? its old comment
do we select staker before tossing coin ?
we toss a coin and select proposer

also this line
```
note that only one staker or no stakers selected in each iteration.
```

Reply:
cool we can remove so, it was never their